test/quantization/fx/test_model_report_fx.py

```diff
@@ -431,3 +435,276 @@ def forward(self, x):
+        model = self.NestedModifiedSingleLayerLinear()
+
+        # generate the desired input
+        ex_input = torch.zeros((10, 1, 5))
+
+        # run it through the model and do general tests
+        self.run_model_and_common_checks(model, ex_input, 1, 1)
+
+        # make sure final values are all 0
+        self.assertEqual(getattr(model, "obs1").epoch_activation_min, 0)
+        self.assertEqual(getattr(model, "obs1").epoch_activation_max, 0)
+        self.assertEqual(getattr(model, "obs1").average_batch_activation_range, 0)
+
+        # we should get an error if we try to calculate the ratio
+        with self.assertRaises(ValueError):
+            ratio_val = getattr(model, "obs1").get_batch_to_epoch_ratio()
+
+    """Case includes:
+    non-zero tensor
+    dim size = 2
+    run for 1 epoch
+    run for 1 batch
+    tests input data observer
+    """
+
+    def test_single_batch_of_ones(self):
+        # initialize the model
+        model = self.NestedModifiedSingleLayerLinear()
+
+        # generate the desired input
+        ex_input = torch.ones((1, 1, 5))
+
+        # run it through the model and do general tests
+        self.run_model_and_common_checks(model, ex_input, 1, 1)
+
+        # make sure final values are all 0 except for range
+        self.assertEqual(getattr(model, "obs1").epoch_activation_min, 1)
+        self.assertEqual(getattr(model, "obs1").epoch_activation_max, 1)
+        self.assertEqual(getattr(model, "obs1").average_batch_activation_range, 0)
+
+        # we should get an error if we try to calculate the ratio
+        with self.assertRaises(ValueError):
+            ratio_val = getattr(model, "obs1").get_batch_to_epoch_ratio()
+
+    """Case includes:
+    non-zero tensor
+    dim size = 2
+    run for 10 epoch
+    run for 15 batch
+    tests non input data observer
+    """
+
+    def test_observer_after_relu(self):
+
+        # model specific to this test
+        class NestedModifiedObserverAfterRelu(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.obs1 = ModelReportObserver()
+                self.mod1 = SingleLayerLinearModel()
+                self.obs2 = ModelReportObserver()
+                self.fc1 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+                self.relu = torch.nn.ReLU()
+
+            def forward(self, x):
+                x = self.obs1(x)
+                x = self.mod1(x)
+                x = self.fc1(x)
+                x = self.relu(x)
+                x = self.obs2(x)
+                return x
+
+        # initialize the model
+        model = NestedModifiedObserverAfterRelu()
+
+        # generate the desired input
+        ex_input = torch.randn((15, 1, 5))
+
+        # run it through the model and do general tests
+        self.run_model_and_common_checks(model, ex_input, 10, 15)
+
+        # make sure final values are all 0
+        self.assertTrue(getattr(model, "obs2").epoch_activation_min >= 0)
```

Comment:
I feel `get_batch_to_epoch_ratio` matches the reference might be stronger than this check?